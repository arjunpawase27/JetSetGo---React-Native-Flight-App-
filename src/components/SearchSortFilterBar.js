import React, { useState } from 'react';
import { View, Button, StyleSheet, Image, TouchableOpacity, Text } from 'react-native';
import SearchComponent from './SearchComponent';

const SearchSortFilterBar = ({ onSort, onClear, onSearch }) => {
  const [showDropdown, setShowDropdown] = useState(false);
  const [selectedSort, setSelectedSort] = useState(null);

  const handleSort = (sortBy) => {
    setShowDropdown(false);
    setSelectedSort(sortBy);
    onSort(sortBy);
  };

  const clearSort = () => {
    setSelectedSort(null);
    onClear();
  };

  return (
    <View style={styles.container}>
      <View style={styles.searchSortContainer}>
        <SearchComponent onSearch={onSearch} />
        <TouchableOpacity onPress={() => setShowDropdown(!showDropdown)}>
          <View style={styles.icon}>
            <Image
              source={require('../assets/images/sort.jpg')} 
              style={styles.iconImage}
            />
          </View>
        </TouchableOpacity>
        
        {showDropdown && (
          <View style={styles.dropdown}>
            <TouchableOpacity onPress={() => handleSort('price')}>
              <Text style={styles.dropdownItem}>Sort by Price</Text>
            </TouchableOpacity>
            <TouchableOpacity onPress={() => handleSort('airline')}>
              <Text style={styles.dropdownItem}>Sort by Airline</Text>
            </TouchableOpacity>
          </View>
        )}
        {/* <TouchableOpacity onPress={onClear}>
          <View style={styles.icon}>
            <Image
              source={require('../assets/images/filter.jpg')} 
              style={styles.iconImage}
            />
          </View>
        </TouchableOpacity> */}
      </View>
      {selectedSort && (
        <TouchableOpacity onPress={clearSort}>
          <View style={styles.chip}>
            <Text style={styles.chipText}>{selectedSort}</Text>
            <Image
              source={require('../assets/images/cancel.jpg')} 
              style={styles.cancelIcon}
            />
          </View>
        </TouchableOpacity>
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginBottom: 20,
  },
  searchSortContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    width: 280,
    //marginHorizontal: 5,
    gap: 5
  },
  icon: {
    width: 45,
    height: 45,
    borderRadius: 25,
    borderWidth: 1,
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: "#ced4da",
    marginBottom: 20
  },
  iconImage: {
    width: 20,
    height: 20,
  },
  dropdown: {
    position: 'absolute',
    top: 50,
    left: 192, 
    backgroundColor: '#ffffff',
    borderWidth: 1,
    borderColor: '#ced4da',
    borderRadius: 10,
    padding: 5,
    zIndex: 999,
  },
  dropdownItem: {
    paddingVertical: 5,
    paddingHorizontal: 10,
    color: "#2f4149"
  },
  chip: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#e9ecef',
    borderRadius: 20,
    paddingHorizontal: 10,
    paddingVertical: 5,
    width:100,
    justifyContent:"space-between",
    marginHorizontal:10
  },
  chipText: {
    marginRight: 5,
    fontWeight:"bold",
    color:"#2f4149"
  },
  cancelIcon: {
    width: 15,
    height: 15,
  },
});

export default SearchSortFilterBar;
