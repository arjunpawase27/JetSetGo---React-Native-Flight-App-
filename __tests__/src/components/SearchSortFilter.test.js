import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import SearchSortFilterBar from '../../../src/components/SearchSortFilterBar';
describe('SearchSortFilterBar', () => {
  test('renders correctly', () => {
    const { toJSON } = render(
      <SearchSortFilterBar
        onSort={() => {}}
        onClear={() => {}}
        onSearch={() => {}}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });

 

 
});
