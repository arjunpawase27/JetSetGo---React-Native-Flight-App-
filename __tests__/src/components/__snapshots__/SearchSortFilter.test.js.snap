// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSortFilterBar renders correctly 1`] = `
<View
  style={
    {
      "marginBottom": 20,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 5,
        "justifyContent": "space-between",
        "width": 280,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#ced4da",
          "borderRadius": 20,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 20,
          "paddingHorizontal": 10,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Search by airline"
        placeholderTextColor="#777"
        style={
          {
            "color": "#495057",
            "flex": 1,
            "height": 40,
            "marginRight": 10,
          }
        }
        value=""
      />
      <Image
        source={
          {
            "testUri": "../../../src/assets/images/search.jpg",
          }
        }
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="dropdown-icon"
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#ced4da",
            "borderRadius": 25,
            "borderWidth": 1,
            "height": 45,
            "justifyContent": "center",
            "marginBottom": 20,
            "width": 45,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/images/sort.jpg",
            }
          }
          style={
            {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
